Bookish.Web/Models/CatalogueViewModel.cs

```diff
@@ -1,15 +1,36 @@
+using System;
 using System.Collections.Generic;
+using System.Linq;
 using Bookish.DataAccess;
 
 namespace Bookish.Web.Models
 {
     public class CatalogueViewModel
     {
         public IEnumerable<CatalogueEntry> Catalogue { get; }
+        public string Filter { get; }
+        public int Page { get; }
+        public int LastPage { get; }
 
-        public CatalogueViewModel(IEnumerable<CatalogueEntry> catalogue)
+        private const int PageSize = 5;
+
+        public CatalogueViewModel(IEnumerable<CatalogueEntry> catalogue, int page = 1, string filter = "")
+        {
+            Page = page;
+            Filter = filter;
+
+            Catalogue = catalogue.ToList();
+            LastPage = (int) Math.Ceiling(Catalogue.Count() / (double) PageSize);
+            Catalogue = Catalogue
```

Comment:
Please avoid reassigning. You should be able to do this by assigning to Catalogue just once 🙂